Library/Services/Implementation/NfieldMediaFilesService.cs

```diff
@@ -110,6 +112,30 @@ public Task AddOrUpdateAsync(string surveyId, string fileName, byte[] content)
                       .FlattenExceptions();
         }
 
+        public Task UploadAndSaveAsync(string surveyId, string fileName, byte[] content)
+        {
+            if (string.IsNullOrEmpty(surveyId))
+            {
+                throw new ArgumentNullException("surveyId");
+            }
+            if (string.IsNullOrEmpty(fileName))
+            {
+                throw new ArgumentNullException("fileName");
+            }
+            if (content == null)
+            {
+                throw new ArgumentNullException("content");
+            }
+
+            var postContent = new ByteArrayContent(content);
+            postContent.Headers.ContentType =
+                new MediaTypeHeaderValue("application/octet-stream");
+
+            return
+               Client.PostAsync(MediaFilesApi(surveyId, fileName), postContent)
+                     .FlattenExceptions();
```

Comment:
We need to make this proper async/await code. So the method should get the async identifier:
public async Task (string surveyId, string fileName, byte[] content)

and the last line should be:
await Client.PostAsync(MediaFilesApi(surveyId, fileName), postContent).FlattenExceptions().ConfigureAwait(false); 

I am not even sure if FlattenExceptions and ConfigureAwait are necessary.